prisma/schema.prisma

```diff
@@ -1,11 +1,74 @@
-// This is your Prisma schema file,
-// learn more about it in the docs: https://pris.ly/d/prisma-schema
-
 generator client {
   provider = "prisma-client-js"
 }
 
 datasource db {
   provider = "postgresql"
   url      = env("DATABASE_URL")
+  shadowDatabaseUrl = env("DATABASE_SHADOW")
+}
+
+model User {
+  id        Int       @id @default(autoincrement())
+  username  String    @unique
+  password  String
+  email     String    @unique
+  createdAt DateTime  @default(now())
+  updatedAt DateTime  @updatedAt
+  genres    Genre[]
+  tags      Tag[]
+  recommendations Recommendation[]
+  authors   Author[]
 }
+
+model Genre {
+  id        Int       @id @default(autoincrement())
+  name      String    @unique
+  createdAt DateTime  @default(now())
+  updatedAt DateTime  @updatedAt
+  users     User[]
+  books     Book[]
+}
+
+model Tag {
+  id        Int       @id @default(autoincrement())
+  name      String    @unique
+  createdAt DateTime  @default(now())
+  updatedAt DateTime  @updatedAt
+  users     User[]
+  books     Book[]
+}
+
+model Book {
+  id              Int               @id @default(autoincrement())
+  title           String        
+  isbn            String            @unique
+  authors         Author[]
```

Comment:
For consistency purposes, I would move the relations to the bottom of the model so that it's easier to read. But I'm being picky here.